client/verta/verta/_registry/modelversion.py

```diff
@@ -70,24 +80,110 @@ def _create_proto_internal(cls, conn, ctx, name, desc=None, tags=None, labels=No
         print("Created new ModelVersion: {}".format(model_version.version))
         return model_version
 
-    def set_model(self, model, overwrite=False):
-        # similar to ExperimentRun.log_artifact
-        raise NotImplementedError
+    def log_model(self, model, overwrite=False):
+        self._refresh_cache()
+        if self.has_model and not overwrite:
+            raise ValueError("model already exists; consider setting overwrite=True")
+
+        if isinstance(model, six.string_types):  # filepath
+            serialized_model = open(model, 'rb')  # file handle
+        else:
+            serialized_model, method, _ = _artifact_utils.serialize_model(model)  # bytestream
+
+        try:
+            extension = _artifact_utils.get_file_ext(serialized_model)
+        except (TypeError, ValueError):
+            extension = _artifact_utils.ext_from_method(method)
+
+        # Create artifact message and update ModelVersion's message:
+        self._msg.model.CopyFrom(self._create_artifact_msg("model", serialized_model, artifact_type=_CommonCommonService.ArtifactTypeEnum.MODEL, extension=extension))
+        self._update()
+
+        # Upload the artifact to ModelDB:
+        self._upload_artifact(
+            "model", serialized_model,
+            _CommonCommonService.ArtifactTypeEnum.MODEL,
+        )
 
     def get_model(self):
-        # similar to ExperimentRun.get_model
-        raise NotImplementedError
+        return _artifact_utils.deserialize_model(self._get_artifact("model", _CommonCommonService.ArtifactTypeEnum.MODEL))
 
-    def add_asset(self, key, asset, overwrite=False):
-        # similar to ExperimentRun.log_artifact
-        raise NotImplementedError
+    def log_artifact(self, key, asset, overwrite=False):
+        if key == "model":
+            raise ValueError("The key `model` is reserved for model. Please use `set_model`")
```

Comment:
Minor wording change based on the convention I've been using for (most of) the Client:
```suggestion
            raise ValueError("the key \"model\" is reserved for model; consider using set_model() instead")
```

Reply:
Will do.

Reply:
Updated.